modules/vpc-serverless-connector-beta/main.tf

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Pending new google-cloud-beta provider release Estimated Release 03/22
+# https://github.com/hashicorp/terraform-provider-google/issues/8475
+resource "google_vpc_access_connector" "connector_beta" {
+  for_each = { for connector in var.vpc_connectors : connector.name => connector }
+  provider = google-beta
+  name     = each.value["name"]
+  project  = var.project_id
+  region   = each.value["region"]
+  subnet {
```

Comment:
nit: we should make this a dynamic block if `each.value["subnet_name"]` exists

Reply:
wont-do: A single subnet(not list) is a required when using the beta therefore is an attribute of the serverless-connector and should always exist when running the module.

Reply:
@tdigangi My concern was wouldn't there be case where the user just wants to specify the ip range and not create a subnet? The usage of the dynamic block would be to conditionally create the subnet block like in https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/1fe315a8ab722d7f91b60e1c209bf938724eb371/modules/simple_bucket/main.tf#L30

Reply:
Added support for these two attributes & dynamic conditional on subnet block